algorithms/dp/longest_common_subsequence.py

```diff
@@ -0,0 +1,22 @@
+def lcs(X , Y): 
+    
+    l1 = len(X) 
```

Comment:
I think we can use a better naming here.
also, could you provide some comment explaining the algo in a high level and what are the types of the arguments?